### ISSUE TITLE
Visualization using Paraview

### ISSUE THREAD

**Issue by lmx08** [GeodynamicWorldBuilder/WorldBuilder#483]
Hey,
I installed version 0.5.0-pre of WB as a stand-alone program on Linux, recently. How can I run the examples in cookbooks  and visualize the result(for example using paraview)?  Thanks for the help!

**Comment by MFraters** [comment]
Hey,
Thanks for your question. When you compiled version 0.5.0-pre, in the bin directory, there should be a binary called `gwb-grid`. It takes two arguments, first a world builder file and secondly a grid file. There is a small section on it in section 3.10 the manual (https://gwb.mfraters.net/manual.pdf), but that definitely need improvement in the new manual. 

More extensive examples of gridfiles can be found in the cookbook folder in the repository. Each cookbook contains a world builder file and a grid file. 

In short, to create vtk output of your world builder model you would do somthing like this in your build directory:
```
./bin/gwb-grid ../cookbook/2d_cartesian_subduction_rift/2d_cartesian_subduction_rift.wb ../cookbook/2d_cartesian_subduction_rift/2d_cartesian_subduction_rift.grid
```

I hope this helps. Let me know if you have any further questions or issues!

**Comment by lmx08** [comment]
Hey, 
Thanks for the answers, this really helps. Recently I tried to install WB on my laptop(macOS, 10.14.6), an error occured as follows:

73/73 Test #73: run_simple_fortran_example .....................***Failed    0.02 sec

86% tests passed, 10 tests failed out of 73

Total Test time (real) =  11.77 sec

The following tests FAILED:
	 59 - run_simple_C_test (Failed)
	 61 - run_simple_C_example (Failed)
	 63 - run_simple_CPP_test (Failed)
	 65 - run_simple_CPP_example (Failed)
	 66 - compile_simple_CPP_MPI_test (Failed)
	 67 - run_simple_CPP_MPI_test (Failed)
	 68 - compile_simple_CPP_MPI_example (Failed)
	 69 - run_simple_CPP_MPI_example (Failed)
	 71 - run_simple_fortran_test (Failed)
	 73 - run_simple_fortran_example (Failed)
Errors while running CTest
Output from these tests are in: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.

Can you help with this? Thanks in advance!

**Comment by MFraters** [comment]
This is a different issue. You have apparently run the tester and all tests pass, except the ones which are related to external interfaces to other programming languages. It is great that you are running the testers, but these specific test failures should not influence running the world builder library or the gwb-grid program.

I would still like to know why they are failing. can you run `ctest --rerun-failed --output-on-failure` and paste or attach the output?

**Comment by lmx08** [comment]
Here is the output:

Test project /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build
      Start 59: run_simple_C_test
 1/10 Test #59: run_simple_C_test ................***Failed    0.05 sec
CMake Error at /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/C/run_C_tests.cmake:32 (message):
  Failed: Test program
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/C/test
  exited != 0 (Subprocess aborted).

   Test args where: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/data/continental_plate.wb
   Error: dyld: Library not loaded: @rpath/libmpi.12.dylib
    Referenced from: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/C/test
    Reason: image not found




      Start 61: run_simple_C_example
 2/10 Test #61: run_simple_C_example .............***Failed    0.02 sec
CMake Error at /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/C/run_C_tests.cmake:32 (message):
  Failed: Test program
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/C/example
  exited != 0 (Subprocess aborted).

   Test args where: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/data/continental_plate.wb
   Error: dyld: Library not loaded: @rpath/libmpi.12.dylib
    Referenced from: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/C/example
    Reason: image not found




      Start 63: run_simple_CPP_test
 3/10 Test #63: run_simple_CPP_test ..............***Failed    0.01 sec
CMake Error at /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/CPP/run_CPP_tests.cmake:32 (message):
  Failed: Test program
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP/test
  exited != 0 (Subprocess aborted).

   Test args where: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/data/continental_plate.wb
   Error: dyld: Library not loaded: @rpath/libmpi.12.dylib
    Referenced from: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP/test
    Reason: image not found




      Start 65: run_simple_CPP_example
 4/10 Test #65: run_simple_CPP_example ...........***Failed    0.01 sec
CMake Error at /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/CPP/run_CPP_tests.cmake:32 (message):
  Failed: Test program
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP/example
  exited != 0 (Subprocess aborted).

   Test args where: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/data/continental_plate.wb
   Error: dyld: Library not loaded: @rpath/libmpi.12.dylib
    Referenced from: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP/example
    Reason: image not found




      Start 66: compile_simple_CPP_MPI_test
 5/10 Test #66: compile_simple_CPP_MPI_test ......***Failed    0.08 sec
/opt/anaconda3/bin/mpicxx: line 283: x86_64-apple-darwin13.4.0-clang++: command not found

      Start 67: run_simple_CPP_MPI_test
 6/10 Test #67: run_simple_CPP_MPI_test ..........***Failed    0.05 sec
CMake Error at /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/CPP_MPI/run_CPP_MPI_tests.cmake:32 (message):
  Failed: Test program
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP_MPI/test
  exited != 0 (255) with mpi /opt/anaconda3/bin/mpiexec.

   Test args where: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/data/continental_plate.wb
   Error: [proxy:0:0@lmx-MacBook-Pro.local] HYDU_create_process (utils/launch/launch.c:74): execvp error on file /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP_MPI/test (No such file or directory)

  [proxy:0:0@lmx-MacBook-Pro.local] HYDU_create_process
  (utils/launch/launch.c:74): execvp error on file
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP_MPI/test
  (No such file or directory)




      Start 68: compile_simple_CPP_MPI_example
 7/10 Test #68: compile_simple_CPP_MPI_example ...***Failed    0.07 sec
/opt/anaconda3/bin/mpicxx: line 283: x86_64-apple-darwin13.4.0-clang++: command not found

      Start 69: run_simple_CPP_MPI_example
 8/10 Test #69: run_simple_CPP_MPI_example .......***Failed    0.03 sec
CMake Error at /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/CPP_MPI/run_CPP_MPI_tests.cmake:32 (message):
  Failed: Test program
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP_MPI/example
  exited != 0 (255) with mpi /opt/anaconda3/bin/mpiexec.

   Test args where: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/data/continental_plate.wb
   Error: [proxy:0:0@lmx-MacBook-Pro.local] HYDU_create_process (utils/launch/launch.c:74): execvp error on file /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP_MPI/example (No such file or directory)

  [proxy:0:0@lmx-MacBook-Pro.local] HYDU_create_process
  (utils/launch/launch.c:74): execvp error on file
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/CPP_MPI/example
  (No such file or directory)




      Start 71: run_simple_fortran_test
 9/10 Test #71: run_simple_fortran_test ..........***Failed    0.02 sec
CMake Error at /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/fortran/run_fortran_tests.cmake:32 (message):
  Failed: Test program
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/fortran/test
  exited != 0.

   Test ars where: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/data/continental_plate.wb

  dyld: Library not loaded: @rpath/libmpi.12.dylib

    Referenced from: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/fortran/test
    Reason: image not found




      Start 73: run_simple_fortran_example
10/10 Test #73: run_simple_fortran_example .......***Failed    0.01 sec
CMake Error at /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/fortran/run_fortran_tests.cmake:32 (message):
  Failed: Test program
  /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/fortran/example
  exited != 0.

   Test ars where: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/tests/data/continental_plate.wb

  dyld: Library not loaded: @rpath/libmpi.12.dylib

    Referenced from: /Users/lmx/Desktop/desktop/world_builder/WorldBuilder-main/build/tests/fortran/example
    Reason: image not found





0% tests passed, 10 tests failed out of 10

Total Test time (real) =   0.37 sec

The following tests FAILED:
	 59 - run_simple_C_test (Failed)
	 61 - run_simple_C_example (Failed)
	 63 - run_simple_CPP_test (Failed)
	 65 - run_simple_CPP_example (Failed)
	 66 - compile_simple_CPP_MPI_test (Failed)
	 67 - run_simple_CPP_MPI_test (Failed)
	 68 - compile_simple_CPP_MPI_example (Failed)
	 69 - run_simple_CPP_MPI_example (Failed)
	 71 - run_simple_fortran_test (Failed)
	 73 - run_simple_fortran_example (Failed)
Errors while running CTest


By the way, during WB installation, when I entered "make release" in the terminal, there are lines of outputs as follows:

-- Found Fortran support. Enabling Fortran wrapper and tests.
-- Could NOT find SWIG (missing: SWIG_EXECUTABLE SWIG_DIR) 
-- SWIG was not found, disabling python wrapper compilation.

I'm not sure if this is related to the errors mentioned above.

**Comment by MFraters** [comment]
Thanks for the output, this has been very helpful. There seems to be an issue with loading mpi on your system.  A few more questions:

1. Are you using a mac M1/2 system?
2. Could you make a new build directory and post the output of `cmake ..`  when run in that directory?
4. Could you try turning off mpi (USE_MPI). You can use `ccmake` to turn it off or `cmake -DUSE_MPI=OFF`, recompile and run the test again. 

In case you are interested, you can turn off Fortran support the same way (WB_MAKE_FORTRAN_WRAPPER). SWIG is used to enable Python support. If you install swig it will find it and enable it (although it can be turned off with WB_ENABLE_PYTHON). Since swig was not found, the python support was turned off, so that is why you don't see any python tests.

**Comment by MFraters** [comment]
hmm, I now see that you are using macOS 10.14.6. This is no longer maintained by Apple since October 2021 (see https://en.wikipedia.org/wiki/MacOS_Catalina). I would strongly recommend that you upgrade your system to the latest version of macos. This will probably solve your problem. You could still try turning off mpi if you don't want to upgrade, but again, I strongly recommend upgrading your system since it hasn't had any security updates since October 2021.

**Comment by lmx08** [comment]
Thanks for the advises, I ran the examples in cookbooks and it works. I'll try to update my mac then.
